numpy/core/tests/test_multiarray.py

```diff
@@ -5846,70 +5846,6 @@ def test_dot_array_order(self):
         assert_equal(np.dot(b, a), res)
         assert_equal(np.dot(b, b), res)
 
-    def test_accelerate_framework_sgemv_fix(self):
```

Comment:
I am +1 on this, but we can still pick up accelerate as a generic cblas implementation. I am not sure we have a way to detect this (it could even happen at import time, not compile time in theory?).
So, I think we should keep this, or other tests around maybe and simply rename it to be clear that someone is using a buggy cblas impelmentation (namely Accelerate).

One option could be to do rename `Accelerate` into `ErrorIfAccelerateIsFound`, which detects Accelerate but then just raises an error. Or alternatively check whether we can check that cblas just points to accelerate.
The user has then the option to:
1. Delete `ErrorIfAccelerateIsFound` to pick up cblas (it is their job to check that this is not accelerate)
2. Use None to use the fallback
3. Install a reasonable BLAS.

We have discussed this a bit, and it seems a bit like we would like something else.

Reply:
In gh-15647 we added a check for Accelerate (well, to be more precise we added a check that the blas backend exhibits the Accelerate bug) at import, which should make this test redundant, no?

Reply:
Right, so we already find all buggy accelerates, and already reject (almost) all accelerate at runtime? I was not sure if the import test covers pretty much all of it. If we assume it does, then I think we should just go there. @WarrenWeckesser does the import time test kick fail for you if you compile with the accelerate as cblas?

Reply:
With this branch or master, I don't get an error on import.  The code added in `_mac_os_check()` in `__init__.py` in doesn't crash when I run it.  However, if I run the full test suite, the unit test `TestPolynomial.test_polyfit` *does* crash, and test crashes if I run just that test.  The following is with Python 3.8.1 and the current master branch of numpy:
```
>>> from numpy.lib.tests.test_polynomial import TestPolynomial
>>> tst = TestPolynomial()
>>> tst.test_polyfit()
python3(18846,0x7fffa26593c0) malloc: *** mach_vm_map(size=18446744073373515776) failed (error code=3)
*** error: can't allocate region
*** set a breakpoint in malloc_error_break to debug
init_dgelsd failed init
<stdin>:1: RankWarning: Polyfit may be poorly conditioned
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/warren/py381/lib/python3.8/site-packages/numpy-1.19.0.dev0+42228fc-py3.8-macosx-10.12-x86_64.egg/numpy/lib/tests/test_polynomial.py", line 171, in test_polyfit
    assert_allclose(mean.std(), 0.5, atol=0.01)
  File "/Users/warren/py381/lib/python3.8/site-packages/numpy-1.19.0.dev0+42228fc-py3.8-macosx-10.12-x86_64.egg/numpy/testing/_private/utils.py", line 1527, in assert_allclose
    assert_array_compare(compare, actual, desired, err_msg=str(err_msg),
  File "/Users/warren/py381/lib/python3.8/site-packages/numpy-1.19.0.dev0+42228fc-py3.8-macosx-10.12-x86_64.egg/numpy/testing/_private/utils.py", line 840, in assert_array_compare
    raise AssertionError(msg)
AssertionError: 
Not equal to tolerance rtol=1e-07, atol=0.01

Mismatched elements: 1 / 1 (100%)
Max absolute difference: 0.5
Max relative difference: 1.
 x: array(0.)
 y: array(0.5)
>>> np.__version__  # Current master branch
'1.19.0.dev0+42228fc'
```

Reply:
For the record, I'm on Mac OS 10.12.6 (Sierra).

Reply:
Also for the record: I edited the code in my previous comment that shows the code that crashes, so the first line is  `import numpy as np`.  If I don't do that, and just start with `from numpy.lib.tests.test_polynomial import TestPolynomial`, the call of `tst.test_polyfit()` doesn't crash.  Go figure.

Reply:
Or can we just replace the current import time test with this one to make it more sensitive to Accelerate? (The alternative I was wondering is whether inspecting the symbols can work.) Otherwise I am still not sure that the `ErrorIfAccelerate`, as ugly and silly as it seems, is actually all that bad.

Reply:
I think we are combining two things here:
- this PR removes the ability to choose Accelerate when building NumPy, and removes code that is meant to try to compensate for a known Accelerate bug
- PR gh-15695 uses a test to detect Accelerate on import. It worked locally for @vrakesh and the successful Azure runs since then confirm it works on CI. I don't know why @WarrenWeckesser's machine can successfully import, but that seems to be an issue related to gh-15695. I have opened a new issue  gh-15946. 

So this PR should wait until the test is improved, and then we can revisit removing the Accelerate-specific code.

Reply:
Yes, agreed. This PR is orthogonal, but on the call there was some consensus that we need to fix/have the test before we should put this in.